File: java/com/google/openlocationcode/tests/RecoverTest.java
package com.google.openlocationcode.tests;

import com.google.openlocationcode.OpenLocationCode;

import junit.framework.Assert;
import org.junit.Test;

/** Test recovery near the poles. */
public class RecoverTest {

    @Test
    public void testRecoveryNearSouthPole() {
        OpenLocationCode olc = new OpenLocationCode("XXXXXX+XX");
        Assert.assertEquals("2CXXXXXX+XX",olc.recover(-81.0,0.0).getCode());
    }

    @Test
    public void testRecoveryNearNorthPole() {
        OpenLocationCode olc = new OpenLocationCode("2222+22");
        Assert.assertEquals("CFX22222+22", olc.recover(89.6, 0.0).getCode());
    }
}

File: kotlin/src/commonTest/kotlin/com/google/openlocationcode/RecoverTest.kt
package com.google.openlocationcode

import kotlin.test.Test
import kotlin.test.assertEquals

class RecoverTest {
    @Test
    fun testRecoveryNearSouthPole() {
        val olc = OpenLocationCode("XXXXXX+XX")
        assertEquals("2CXXXXXX+XX", olc.recover(-81.0, 0.0).code)
    }

    @Test
    fun testRecoveryNearNorthPole() {
        val olc = OpenLocationCode("2222+22")
        assertEquals("CFX22222+22", olc.recover(89.6, 0.0).code)
    }
}
